ide-common/src/main/java/org/digma/intellij/plugin/document/CodeObjectsUtil.java

```diff
@@ -2,9 +2,36 @@
 
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Collection;
+import java.util.List;
+
 public class CodeObjectsUtil {
 
     public static String createSpanId(@NotNull String instrumentationLibrary, @NotNull String name) {
         return instrumentationLibrary + "$_$" + name;
     }
+
+    public static String createSpanIdWithType(@NotNull String instrumentationLibrary, @NotNull String name) {
+        return "span:"+createSpanId(instrumentationLibrary,name);
+    }
+
+    public static String createMethodCodeObjectId(@NotNull String className, @NotNull String methodName) {
+        return className + "$_$" + methodName;
+    }
+
+    public static String createMethodCodeObjectIdWithType(@NotNull String className, @NotNull String methodName) {
+        return "method:"+createMethodCodeObjectId(className,methodName);
+    }
+
+    public static List<String> addSpanTypeToIds(@NotNull Collection<String> spanIds) {
+        return spanIds.stream().map(id -> "span:"+id).toList();
+    }
+
+    public static List<String> addMethodTypeToIds(@NotNull Collection<String> methodIds) {
```

Comment:
I would also check 
if spanId is already typed (has any type prefix){
     if type not equals to method {
           throw exception
      }
}
else{
add "method" prefix
}